lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js

```diff
@@ -123,21 +123,28 @@ describe('SubmitGradeForm', () => {
   });
 
   context('validation messages', () => {
+    beforeEach(() => {
+      $imports.$mock({
+        '../utils/validation': {
+          validateGrade: sinon.stub().returns('err'),
+        },
+      });
+    });
+
     it('does not render the validation message by default', () => {
       const wrapper = renderForm();
       assert.isFalse(wrapper.find('ValidationMessage').exists());
     });
 
     it('shows the validation message when the validator returns an error', () => {
-      $imports.$mock({
-        '../utils/validation': {
-          validateGrade: sinon.stub().returns('err'),
-        },
-      });
       const wrapper = renderForm();
       wrapper.find('button[type="submit"]').simulate('click');
       assert.isTrue(wrapper.find('ValidationMessage').prop('open'));
       assert.equal(wrapper.find('ValidationMessage').prop('message'), 'err');
+
+      // Clicking the error message should dismiss it.
+      wrapper.find('ValidationMessage').invoke('onClose')();
```

Comment:
I've never encountered `wrapper.invoke` before (though it's totally there in Enzyme's docs). Interesting! 

Also, I _feel_ like this is one of those tests that you'd nudge me about using `act` for. Can you help me understand why it's not the proper convention here?

Reply:
`wrapper.invoke` is a recent addition to Enzyme. It is effectively shorthand for:

```js
act(() => {
  wrapper.prop(<propName>)(...args);
});
wrapper.update();
```